torch/_dynamo/symbolic_convert.py

```diff
@@ -581,6 +581,26 @@ def handle_graph_break(
     return decorator
 
 
+def promote_exception_to_graph_break(inner_fn):
+    @functools.wraps(inner_fn)
+    def wrapper(self: "InstructionTranslatorBase", inst: Instruction):
+        try:
+            TracingContext.set_current_loc(
+                self.f_code.co_filename, self.lineno, self.f_code.co_name
+            )
+            return inner_fn(self, inst)
+        except Unsupported:
+            raise
+        except RuntimeError as e:
```

Comment:
Without this PR, Dynamo will convert the RuntimeError into either a InternalTorchDynamoError or TorchRuntimeError. We should only convert RuntimeErrors that would have been converted into TorchRuntimeError into graph breaks; InternalTorchDynamoError RuntimeErrors are Dynamo bugs and should not be silently ignored.

(that being said I am not sure where this logic is)